server/src/lib/model/db.ts

```diff
@@ -42,9 +42,20 @@ export default class DB {
   /**
    * Insert or update user client row.
    */
-  async updateUser(client_id: string, fields: any): Promise<void> {
+  async updateUser(
+    client_id: string,
+    fields: any,
+    bucket = 'train'
+  ): Promise<void> {
     let { age, accent, email, gender } = fields;
     email = this.formatEmail(email);
+    await this.mysql.query(
+      `
+        INSERT INTO user_clients (client_id, bucket) VALUES (?, ?)
+        ON DUPLICATE KEY UPDATE client_id = client_id
+      `,
+      [client_id, bucket]
+    );
```

Comment:
Hrm, this adds an additional upsert every time we want to update the user. This is confusing and also spreads the user_client queries between separate code modules (db, and userClient). Let's make this a part of the userClient update call instead (which will require some more robust logic to determine when to reset the bucket, which I think will also be good for code readability and quality).

Reply:
I put it outside of the `userClient.update` because every other field in there can be changed, while the bucket should be only set on creation. (Maybe I should add a comment explaining that reasoning).
I guess I could also overwrite `userClient.update` and just move the call into it. Would that be what you have in mind?

Reply:
yeah, probably better to have both queries there (if two queries are unavoidable)